runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -91,6 +96,20 @@
     @Getter
     private Set<UUID> streamsToIgnore = new HashSet<>();
 
+    /**
+     * We can register streams with non-default type
+     */
+    @Getter
+    private Map<UUID, Class> nonDefaultTypeStreams = new HashMap<>();
```

Comment:
how about a compromise: add this method before merging, even though MP will not yet use it?